### PR TITLE
feat: route settlement through aibtc relay

### PR DESCRIPTION
**Pull request by whoabuddy**
## Summary

- Switch `X402_FACILITATOR_URL` from external facilitator (`facilitator.stacksx402.com`) to aibtc relay (`x402-relay.aibtc.com` for prod, `x402-relay.aibtc.dev` for staging)
- Rename `FACILITATOR_UNAVAILABLE` → `RELAY_UNAVAILABLE` and `FACILITATOR_ERROR` → `SETTLEMENT_ERROR` in error classification
- Fix classification bugs: `TRANSACTION_PENDING` no longer falls through to `UNKNOWN_ERROR`, `UNSUPPORTED_SCHEME` correctly classified as `PAYMENT_INVALID`, operator precedence bug fixed
- Update toolbox UI link and test utility retryable codes to match new names
- Remove dead `facilitator` variable from `renderPaymentDetails`

Closes #43

## Test plan

- [ ] Verify build compiles cleanly (`wrangler deploy --dry-run`)
- [ ] Confirm no stale facilitator references in source (env var name `X402_FACILITATOR_URL` intentionally preserved)
- [ ] Test payment flow against relay staging endpoint
- [ ] Verify error classification for TRANSACTION_PENDING, UNSUPPORTED_SCHEME edge cases

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/endpoints/toolbox.ts [hunk 1]
```diff
@@ -704,7 +702,7 @@ function generateToolboxHTML(): string {
       html += '<h3>Payment Details</h3>';
       html += '<div class="info-row"><span class="info-label">Pay To</span><span class="info-value">' + (payTo ? escapeHtml(payTo.slice(0,12) + '...' + payTo.slice(-8)) : 'Unknown') + '</span></div>';
       html += '<div class="info-row"><span class="info-label">Network</span><span class="info-value">' + (network || 'mainnet') + '</span></div>';
-      html += '<div class="info-row"><span class="info-label">Facilitator</span><span class="info-value"><a href="https://facilitator.x402stacks.xyz" target="_blank">facilitator.x402stacks.xyz</a></span></div>';
+      html += '<div class="info-row"><span class="info-label">Relay</span><span class="info-value"><a href="https://x402-relay.aibtc.com" target="_blank">x402-relay.aibtc.com</a></span></div>';
       html += '</div>';
```
[the remark this answers is not in the document]
Comment (whoabuddy) [reply src/endpoints/toolbox.ts]: Fixed in 6d910f3 — relay URL is now derived from `facilitatorUrl` in the payment requirements instead of being hardcoded. Row only renders when the field is present.

File: wrangler.jsonc [hunk 2]
```diff
@@ -43,7 +43,7 @@
   ],
   // Environment variables for X402 payment configuration
   "vars": {
-    "X402_FACILITATOR_URL": "https://facilitator.stacksx402.com", // X402 payment facilitator endpoint
+    "X402_FACILITATOR_URL": "https://x402-relay.aibtc.com", // X402 relay endpoint (routes settlement via aibtc relay)
```
[the remark this answers is not in the document]
Comment (whoabuddy) [reply wrangler.jsonc]: Acknowledged. The patch file only contains "facilitator" in upstream library context lines (not lines we control), so no update was needed. PR description was misleading — the phase 1 investigation found no actionable patch changes. Updated description would clarify this.
